### DISCUSSION TITLE
Schema / Data Table Model Export

### DISCUSSION THREAD

**Discussion by jtkeyva** [supabase/supabase#9616] [Feature Requests]
# Feature request
Schema / Data model export
## Is your feature request related to a problem? Please describe.

A clear and concise description of what you want and what your use case is.
A model export button that provides a list and or json of the table column/field names and types. An easy time saver and typo prevention for creating model classes in say Flutter.
## Describe the solution you'd like
A button with popup that has the data displayed in a list and or json format and a copy to clipboard button. This could be per table or for the entire database. Could also provide export options for different languages ex Javascript, Flutter, etc.
Get a perfect replica of your table models with data types and their properties ex:

Uuid id //required
String username //unique //required
DateTime created_at
Int index //autoincrement

## Describe alternatives you've considered
someone made a flutter package that creates a model from a supabase db
https://github.com/osaxma/schema-dart/

**Comment by salazarr-js** [comment]
something similar to `supabase-schema`?
https://github.com/zernonia/supabase-schema
https://supabase-schema.vercel.app/

**Comment by kiwicopple** [comment]
We also have generators for Typescritpt: https://supabase.com/docs/reference/javascript/typescript-support

> Get a perfect replica of your table models with data types and their properties

It's worth pointing out that your database is _just_ postgres, so you can also run a pg_dump on the database to get a full database schema which you can use to recreate another Postgres database

**Reply by jtkeyva** [reply]
@kiwicopple thanks, a Model generator for Flutter would be amazing :)

**Reply by mmvergara** [reply]
> @kiwicopple thanks, a Model generator for Flutter would be amazing :)

Hey i don't know if you still need it, but here's one... kinda, but maybe it can help
https://github.com/mmvergara/supabase-schema-dart-class-generator

### Fetch Data

```dart
// fetchedBooks is a typeof List<Books>
final books = await supabase
      .books
      .select("*")
      .withConverter((data) => data.map(Books.fromJson).toList());
```

**Comment by Revadike** [comment]
I'd love a JSON schema export tool for the database
https://json-schema.org/
